subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/AbstractCrossVersionPerformanceTest.groovy

```diff
@@ -42,6 +43,9 @@ class AbstractCrossVersionPerformanceTest extends Specification {
     final CrossVersionPerformanceTestRunner runner = new CrossVersionPerformanceTestRunner(
         new BuildExperimentRunner(new GradleSessionProvider(buildContext)), resultStore, new ReleasedVersionDistributions(buildContext), buildContext)
 
+    @Rule
+    PerformanceTestIdProvider performanceTestIdProvider = new PerformanceTestIdProvider(runner)
```

Comment:
Actually we should add it to all `Abstract*PerformanceTest`s in `org.gradle.performance`. I totally missed that. 

I made the rule more generic and added it to all abstract performance tests